File: ConcursInot/src/main/java/ro/mpp2024/Repository/Repository.java
package ro.mpp2024.Repository;

import ro.mpp2024.domain.Entity;

public interface Repository <ID, E extends Entity<ID>> {
    E findById(ID id);
    Iterable<E> findAll();
    void save(E entity);
    void update(E entity);
    void delete(ID id);
}

File: ConcursInotCS/ConcursInotCS/Repository/Repository.cs
using ConcursInotCS.domain;

namespace ConcursInotCS.Repository;

public interface Repository<ID,E> where E : Entity<ID>
{
    E findById(int id);
    IEnumerable<E> findAll();
    E save(E entity);
    E update(E entity);
    E delete(E entity);
}